drivers/hmis/app/models/hmis/form/processor.rb

```diff
@@ -0,0 +1,196 @@
+
+class Hmis::Form::Processor < ::GrdaWarehouseBase
+  self.table_name = :hmis_assessment_processors
+
+  has_one :assessment_detail
+
+  # assessment is accessed through the assessment_detail
+  belongs_to :health_and_dv, class_name: 'Hmis::Hud::HealthAndDv', primary_key: :HealthAndDVID, optional: true
+  belongs_to :income_benefit, class_name: 'Hmis::Hud::IncomeBenefit', primary_key: :IncomeBenefitID, optional: true
+  belongs_to :enrollment_coc, class_name: 'Hmis::Hud::EnrollmentCoc', primary_key: :EnrollmentCocID, optional: true
+  belongs_to :physical_disability, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilityID, optional: true
+  belongs_to :developmental_disability, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilitiesID, optional: true
+  belongs_to :chronic_health_condition, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilitiesID, optional: true
+  belongs_to :hiv_aids, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilitiesID, optional: true
+  belongs_to :mental_health_disorder, class_name: 'Hmis::Hud::Disability', primary_key: :DisabilitiesID, optional: true
+  belongs_to :substance_use_disorder, class_name: 'Hmis::Hud::Disability', primary_key: :DisabiliesID, optional: true
+
+  def run!
+    return unless assessment_detail.hud_values.present?
+
+    assessment_detail.hud_values.each do |key, value|
+      # Don't use greedy matching so that the container is up to the first dot, and the rest is the field
+      container, field = /(.*?)\.(.*)/.match(key)[1..2]
+
+      container_processor(container)&.process(field, value)
+    end
+  end
+
+  # Type Factories
+  def enrollment_factory
+    # The enrollment has already been created, so we can just return it
+    assessment_detail.assessment.enrollment
+  end
+
+  # The items associated with the enrollments are all singletons, so return
+  # them if they already exist, otherwise create them
+  def enrollment_coc_factory
+    return enrollment_coc if enrollment_coc.present?
+
+    self.enrollment_coc = enrollment_factory.enrollment_cocs.
+      where(data_collection_stage: assessment_detail.data_collection_stage).
+      first_or_initialize(
+        enrollment_id: enrollment_factory.enrollment_id,
+        household_id: enrollment_factory.household_id,
+        project_id: enrollment_factory.project_id,
+        personal_id: enrollment_factory.client.personal_id,
+        information_date: assessment_detail.assessment.assessment_date,
+        user_id: assessment_detail.assessment.user_id,
+      )
+  end
+
+  def health_and_dv_factory
+    return health_and_dv if health_and_dv.present?
+
+    self.health_and_dv = enrollment_factory.health_and_dvs.
+      where(data_collection_stage: assessment_detail.data_collection_stage).
+      first_or_initialize(
+        enrollment_id: enrollment_factory.enrollment_id,
+        personal_id: enrollment_factory.client.personal_id,
+        information_date: assessment_detail.assessment.assessment_date,
```

Comment:
The information date will be coming from hudValues. Either as a single `informationDate` key (preferable to reduce form definition complexity) or as multiple fields like `IncomeBenefit.informationDate` etc. 

I think we should probably store the InformationDate directly on the AssessmentDetail too, so that it can be used as a column in the assessments table for example.

Reply:
After discussion, we decided this will come from the assessment as implemented, and additionally, we need to update the `information_date` to match the `assessment_date` whenever it changes.